### PR TITLE
DB data saving, loading and deleting in loops, others MEQP fixes.

### PR DESCRIPTION
**Pull request by giorello**
Fixes for:
- methods save(), delete(), load(), etc, in loops.
- Lines exceeding 120 characters.
- Count() in loops.
- Others.

Ref, #779 .

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 1]
```diff
@@ -2455,7 +2485,7 @@ protected function _makeForCollectionItem($collection, $mailchimpStoreId, $initi
 
         do {
             $collection->setCurPage($currentPage);
-            $collection->load();
+            $this->_loadItemCollection($collection);
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Unnecessary, please remove entire function.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 2]
```diff
@@ -2477,6 +2507,14 @@ protected function _makeForCollectionItem($collection, $mailchimpStoreId, $initi
...
     }
 
+    /**
+     * @param $collection
+     */
+    protected function _loadItemCollection(&$collection)
+    {
+        $collection->load();
+    }
+
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Unnecessary, please remove entire function.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 3]
```diff
@@ -2813,11 +2851,21 @@ protected function _deletedWebhooksByListId($api, $listId, $webhookUrl)
         $webhooks = $api->lists->webhooks->getAll($listId);
         foreach ($webhooks['webhooks'] as $webhook) {
             if (strpos($webhook['url'], $webhookUrl) !== false) {
-                $api->lists->webhooks->delete($listId, $webhook['id']);
+                $this->_deleteWebhookFromList($api->lists->webhooks, $listId, $webhook['id']);
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Change with $api->getLists()->getWebhooks()

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 4]
```diff
@@ -5024,8 +5072,18 @@ public function ping($storeId)
...
+        $itemType,
+        $syncDelta,
+        $syncError,
+        $syncModified,
+        $syncDeleted,
+        $token,
+        $syncedFlag,
+        $deletedRelatedId,
+        $allowBatchRemoval,
+        Varien_Object $ecommerceSyncDataItem
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Incorrect class name

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Batches.php [hunk 5]
```diff
@@ -757,9 +771,8 @@ protected function processEachResponseFile($files, $batchId, $mailchimpStoreId,
                     $type = $line[1];
                     $id = $line[3];
 
-                    if ($item['status_code'] != 200) {
+                    if ($item->status_code != 200) {
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Batches.php]: Why reverting code standard?

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Batches.php [hunk 6]
```diff
@@ -863,6 +877,7 @@ protected function _getError($type, $mailchimpStoreId, $id, $response)
     protected function _processFileErrors($response)
     {
         $errorDetails = "";
+        $errors = $response->errors;
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Batches.php]: $response should be handled as array (check coding standards)

**Review by SantiParagarino — CHANGES_REQUESTED** [review]

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php (hunk 4 above)
Comment (giorello) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Solved in Merge with develop.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php (hunk 1 above)
Comment (giorello) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: This load is necessary to load the next page (filter) in the collection using the same object. If we remove the Load method of that collection, the page iterated would be always the same.

**Review by SantiParagarino — APPROVED** [review]

**Pull request merged**